### ISSUE TITLE
Update drop highlighting with source/target validation logic

### ISSUE THREAD

**Issue by bfitzpat** [fusesource/fuseide#250]
Discovered a bug with the drag/drop validation done with #218. Dragging/dropping from root element in tree to detail/transformations pane fields causes a NPE (see below). 

But also had a chat with @jpav about doing validation for the drops in the detail and transformations panes as well as when dragging/dropping between the tree viewers. Need to address it in the "valid" method of the PotentialDropTarget instances to get that working correctly. 

```
!ENTRY org.jboss.tools.fuse.transformation.editor 4 0 2015-04-27 16:05:37.839
!MESSAGE Unexpected error: null
!STACK 0
java.lang.NullPointerException
	at org.jboss.tools.fuse.transformation.dozer.DozerMapperConfiguration.getClassMapping(DozerMapperConfiguration.java:432)
	at org.jboss.tools.fuse.transformation.dozer.DozerMapperConfiguration.requiresClassMapping(DozerMapperConfiguration.java:350)
	at org.jboss.tools.fuse.transformation.dozer.DozerMapperConfiguration.mapField(DozerMapperConfiguration.java:237)
	at org.jboss.tools.fuse.transformation.dozer.DozerMapperConfiguration.mapField(DozerMapperConfiguration.java:41)
	at org.jboss.tools.fuse.transformation.editor.internal.util.TransformationConfig.update(TransformationConfig.java:487)
	at org.jboss.tools.fuse.transformation.editor.internal.util.TransformationConfig.setTarget(TransformationConfig.java:451)
	at org.jboss.tools.fuse.transformation.editor.internal.MappingViewer.dropOnTarget(MappingViewer.java:128)
	at org.jboss.tools.fuse.transformation.editor.internal.MappingViewer$3.drop(MappingViewer.java:91)
	at org.jboss.tools.fuse.transformation.editor.internal.MappingViewer$DropListener.drop(MappingViewer.java:204)
	at org.eclipse.swt.dnd.DNDListener.handleEvent(DNDListener.java:91)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:84)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4353)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1061)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1085)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1070)
	at org.eclipse.swt.widgets.Widget.notifyListeners(Widget.java:782)
	at org.eclipse.swt.dnd.DropTarget.Drop(DropTarget.java:456)
	at org.eclipse.swt.dnd.DropTarget.Drop_64(DropTarget.java:392)
	at org.eclipse.swt.dnd.DropTarget$3.method6(DropTarget.java:260)
	at org.eclipse.swt.internal.ole.win32.COMObject.callback6(COMObject.java:119)
	at org.eclipse.swt.internal.ole.win32.COM.DoDragDrop(Native Method)
	at org.eclipse.swt.dnd.DragSource.drag(DragSource.java:363)
	at org.eclipse.swt.dnd.DragSource.access$0(DragSource.java:289)
	at org.eclipse.swt.dnd.DragSource$1.handleEvent(DragSource.java:172)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:84)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4353)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1061)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4172)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3761)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$9.run(PartRenderingEngine.java:1151)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:332)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1032)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:148)
	at org.eclipse.ui.internal.Workbench$5.run(Workbench.java:636)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:332)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:579)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:150)
	at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:135)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:380)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:235)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:648)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:603)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1465)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1438)
```

**Comment by bfitzpat** [comment]
Still working on this. There are some interesting combinations of code to validate the way things are set up.

**Comment by jpav** [comment]
Fixed by issue #252
